src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceLoggerExtension.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics.Extensions
+{
+    public static class ScriptHostServiceLoggerExtension
+    {
+        private static readonly string _codeArea = nameof(ScriptHostServiceLoggerExtension).Replace("LoggerExtension", string.Empty);
+
+        private static readonly EventId _scriptHostServiceInitCanceledByRuntime = new EventId(0, $"{_codeArea}\\{nameof(ScriptHostServiceInitCanceledByRuntime)}");
+        private static readonly EventId _unehealthyCountExceeded = new EventId(0, $"{_codeArea}\\{nameof(UnhealthyCountExceeded)}");
+        private static readonly EventId _offline = new EventId(0, $"{_codeArea}\\{nameof(Offline)}");
+        private static readonly EventId _initializing = new EventId(0, $"{_codeArea}\\{nameof(Initializing)}");
+        private static readonly EventId _initialization = new EventId(0, $"{_codeArea}\\{nameof(Initialization)}");
+        private static readonly EventId _inStandByMode = new EventId(0, $"{_codeArea}\\{nameof(InStandByMode)}");
+        private static readonly EventId _unhealthyRestart = new EventId(0, $"{_codeArea}\\{nameof(UnhealthyRestart)}");
+        private static readonly EventId _stopping = new EventId(0, $"{_codeArea}\\{nameof(Stopping)}");
+        private static readonly EventId _didNotShutDown = new EventId(0, $"{_codeArea}\\{nameof(DidNotShutDown)}");
+        private static readonly EventId _shutDownCompleted = new EventId(0, $"{_codeArea}\\{nameof(ShutDownCompleted)}");
+        private static readonly EventId _skipRestart = new EventId(0, $"{_codeArea}\\{nameof(SkipRestart)}");
+        private static readonly EventId _restarting = new EventId(0, $"{_codeArea}\\{nameof(Restarting)}");
+        private static readonly EventId _restarted = new EventId(0, $"{_codeArea}\\{nameof(Restarted)}");
+        private static readonly EventId _building = new EventId(0, $"{_codeArea}\\{nameof(Building)}");
+        private static readonly EventId _startupWasCanceled = new EventId(0, $"{_codeArea}\\{nameof(StartupWasCanceled)}");
+        private static readonly EventId _errorOccured = new EventId(0, $"{_codeArea}\\{nameof(ErrorOccured)}");
+        private static readonly EventId _errorOccuredInactive = new EventId(0, $"{_codeArea}\\{nameof(ErrorOccuredInactive)}");
+        private static readonly EventId _cancellationRequested = new EventId(0, $"{_codeArea}\\{nameof(CancellationRequested)}");
```

Comment:
A couple comments:
- We should use something other than 0 here. Even though we don't use the integer, other logger providers may.
- Doing something like `_codeArea` shouldn't be needed. CategoryName + EventId is a unique identifier. Effectively, it's already "namespaced" because it'll be logged by a specific logger category. I think this is where I first read it -- https://github.com/aspnet/Logging/issues/628

See here for an example of how ASP.NET sets some of their logging up: https://github.com/aspnet/AspNetCore/blob/f2e6e6ff334176540ef0b3291122e359c2106d1a/src/Security/Authentication/Core/src/LoggingExtensions.cs

Reply:
If we want EventId we need to add new column to the Kusto table. Existing EventId column is already used. 
I agree we can remove `_codeArea`